design/datamover.md

```diff
@@ -0,0 +1,48 @@
+- We also want to allow vendors to provide their own volume data management strategies to be able to work within the velero backup/restore workflow. 
+
+## What is a Data Mover ?
+Data Mover defines an interface that enables movement of kubernetes data between backup storage.
+
+## Goals
+- The data mover should be **Pluggable**. This means that a user can leverage different types of movers for the same set of resources as long as the movers conform to a similar interface.
+- The interface for Data Mover should support moving data associated with resources **other than** Persistent Volumes
+  - Persistent Volumes are an immediate need, and will likely drive most of the original design discussions.
+  - Ex. a Data Mover for Kubevirt would move VM data to backup storage. The interface should be generic enough to support this use case.
+- The interface for Data Mover should enable publishing it’s specific configuration parameters for users to have fine-grained control over the movement workflow
+  - Examples:
+    - Data mover for vsphere volume can configure block vs file mount usage
+    - Kubevirt mover can configure export file type
+    - Image mover can specify image format type
+- End-user interaction with data mover should be exposed through the existing backup/restore operations
+
+## Non-Goals
+- This implementation will **NOT** change the existing Velero backup/restore workflow.
+  - It is possible in the future that this interface can be used to replace some legacy Velero implementations, but that is **NOT** a goal for this design/implementation.
+- This design of the interface should not include any specific details about (aside from samples):
+  - A mover type
+    - Although we are focused on volumes for phase 1, the interface should have zero knowledge of the underlying data that needs to be moved.
+  - Mover implementations
+    - Ex. How a mover handles volumes at the block vs filesystem level is not relevant to the overarching interface design
+
+## Use Cases
+- Data Mover for Persistent Volume Snapshots
+  - For a given snapshot type, a capable data mover should be able to clone the snapshot data into some backup target.
+  - The most urgent use case for this today is CSI snapshots of volumes where the given CSI driver does not guarantee durability and portability. A user creates a CSI snapshot of a volume and wishes to clone this snapshot data into some backup target. The user then would like to be able restore a volume from this snapshot that exists in the backup target.
+- User wants to consume third party vendor data movers
+  - Example: Dell Power Protect has their own data mover to handle PVs and would like to integrate this into the native Velero backup workflow as opposed to it being a separate and disconnected process.To do this, Velero’s data mover interface has to support the ability for a vendor to bring their own data mover and have the Velero backup/restore workflow trigger it’s process.
```

Comment:
Since Dell PowerProtect does various additional activities like quiescing application pods through application templates before taking CSI snapshot, custom support for storage that support volumegroup snapshots CRD etc, I am not sure it can switch to Velero for CSI snapshots. In any case,  datamovement should be optional and controllable as part of backup request or Velero deployment.